packages/utils/src/misc/merge-env-with-options.ts

```diff
@@ -0,0 +1,116 @@
+import { merge } from 'lodash';
+
+const isWebhintVariable = (variable: string) => {
+    return variable.toLowerCase().startsWith('webhint_');
+};
+
+const parseValue = (value: string): string | number | boolean => {
+    const parsedInt = parseInt(value);
+
+    if (!isNaN(parsedInt)) {
+        return parsedInt;
+    }
+
+    if (value === 'true') {
+        return true;
+    }
+
+    if (value === 'false') {
+        return false;
+    }
+
+    return value;
+};
+
+/**
+ * Adds the value of a variable to an object creating as many properties as
+ * needed.
+ *
+ * E.g.:
+ *
+ * ```js
+ * addToConfig('webhint_connector_options_auth_username', 'johndoe', {});
+ * ```
+ *
+ * gets transformed into
+ *
+ * ```json
+ * {
+ *   "connector": {
+ *     "options": {
+ *       "auth": {
+ *         "username": "johndoe"
+ *       }
+ *     }
+ *   }
+ * }
+ * ```
+ *
+ * @param variable The variable to add to a config. E.g.:
+ *   `webhint_connector_options_auth_username`
+ * @param value Any string
+ * @param config The object to add the config too
+ */
+const addToConfig = (variable: string, value: string, config: any) => {
+    /** `variable` example: webhint_connector_options_auth_username */
+    const parts = variable.split('_');
+
+    const parsedValue = parseValue(value);
+
+    /** First item is `webhint` and it is not needed */
+    parts.shift();
+
+    let current = config;
```

Comment:
Or if you prefer, I think `reduce` would be a nice way to replace the whole `while` loop:
```suggestion
    const lastPart = parts.pop();
    const context = parts.reduce((current, part) => {
        /* Create the property if necessary */
        return (current[part] = current[part] || {});
    }, config);
    
    context[lastPart] = parseValue(value);
```